compiler/utils/arena/src/lib.rs

```diff
@@ -1,7 +1,53 @@
 #![forbid(unsafe_code)]
-#![warn(unused_crate_dependencies, unused_extern_crates)]
```

Comment:
Why is this removed?

Reply:
There are now zero dependencies in this crate, but if you think we might add more later and this would be useful I'll revert.

Reply:
Ah gotcha. I'd rather just leave it in. If we can keep this stack consistent across all the other crates then it's easy to just copy-paste when we create a new one.